From 0bfca2d07168b6959fe97d115b8c48db1ce9132c Mon Sep 17 00:00:00 2001
From: Lucas Colley <lucas.colley8@gmail.com>
Date: Thu, 25 Sep 2025 21:24:29 +0100
Subject: [PATCH 4/4] avoid stripping out C stdlib too zealously

this avoids failures on macOS 26 like `must link with at least
libSystem.dylib`
---
 cctools/ld64/src/ld/passes/dylibs.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cctools/ld64/src/ld/passes/dylibs.cpp b/cctools/ld64/src/ld/passes/dylibs.cpp
index 8f8ee46..d129024 100644
--- a/cctools/ld64/src/ld/passes/dylibs.cpp
+++ b/cctools/ld64/src/ld/passes/dylibs.cpp
@@ -306,7 +306,7 @@ void doPass(const Options& opts, ld::Internal& state)
 		if ( aDylib->explicitlyLinked() && aDylib->deadStrippable() && !aDylib->providedExportAtom() && !aDylib->neededDylib() )
 			aDylib->setWillBeRemoved(true);
 		// set "willRemoved" bit on any unused explicit when -dead_strip_dylibs is used
-		if ( opts.deadStripDylibs() && !aDylib->providedExportAtom() && !aDylib->neededDylib() )
+		if ( opts.deadStripDylibs() && !aDylib->providedExportAtom() && !aDylib->neededDylib() && strncmp(aDylib->installPath(), "/usr/lib/libSystem.", 19) != 0 )
 			aDylib->setWillBeRemoved(true);
 		// <rdar://problem/48642080> Warn when dylib links itself
 		if ( (opts.outputKind() == Options::kDynamicLibrary) && !aDylib->willRemoved() ) {
